packages/go_router/lib/src/delegate.dart

```diff
@@ -278,6 +282,7 @@ class ImperativeRouteMatch extends RouteMatch {
     required super.error,
     required super.pageKey,
     required this.matches,
+    super.completer,
```

Comment:
Somewhere in the doc should mention about the completer and future and what it meant.

also, please remove the todo for me. Thanks in advance

Reply:
Done